gnovm/pkg/gnolang/ownership.go

```diff
@@ -376,3 +417,69 @@ func (tv *TypedValue) GetFirstObject(store Store) Object {
 		return nil
 	}
 }
+
+// GetFirstObjectAndAttach returns the value of [TypedValue.GetFirstObject], but
+// additionally sets its boundRealm
+func (tv *TypedValue) GetFirstObjectAndAttach(store Store) Object {
+	obj := tv.GetFirstObject(store)
+	if obj == nil {
+		// no object to set the bound realm to.
+		return nil
+	}
+
+	var path string
+	// infer original package using declared type
+	getPkgId := func(t Type) (pkgId PkgID) {
+		if dt, ok := t.(*DeclaredType); ok {
+			path = dt.GetPkgPath()
+			pkgId = PkgIDFromPkgPath(dt.GetPkgPath())
+		}
+		return
+	}
+
+	var boundRealm PkgID
+
+	switch cv := obj.(type) {
+	case *HeapItemValue:
+		boundRealm = getPkgId(cv.Value.T)
+	case *Block:
+		// assert to pointer value
+		if pv, ok := tv.V.(PointerValue); ok {
+			boundRealm = getPkgId(pv.TV.T)
```

Comment:
if tv.V.(PointerValue), and tv.GetFirstObject() is *Block, 
pv.TV is the thing being pointed to, 
but getPkgId(pv.TV.T) has no relation to the block obj.

attachment comes from the LHS, all this logic is coming from the RHS